### PR TITLE
Document rare segmentation fault triggered by Ruby profiler

### PR DESCRIPTION
**Pull request by ivoanjo**
### What does this PR do? What is the motivation?

This PR documents an issue triggered by the Ruby profiler in certain Ruby VM versions (see
<https://github.com/DataDog/dd-trace-rb/issues/3430> and <https://bugs.ruby-lang.org/issues/19991> for reference), as well as the current recommended solution for this issue.

### Merge instructions

- [x] Please merge after reviewing

### Additional notes

N/A

### CODE REVIEW

File: content/en/profiler/profiler_troubleshooting/ruby.md [hunk 1]
```diff
@@ -73,6 +73,20 @@ end
 Let our team know if you find or suspect any incompatibilities [by opening a support ticket][2].
 Doing this enables Datadog to add them to the auto-detection list, and to work with the gem/library authors to fix the issue.
 
+## Segmentation faults in `gc_finalize_deferred` in Ruby versions 2.6 to 3.2
+
+In rare situations, we've discovered that the profiler can trigger [Ruby VM Bug #19991][12] that manifests itself as a "Segmentation fault" with a crash stack trace including the `gc_finalize_deferred` function.
```
Comment (aliciascott) [inline content/en/profiler/profiler_troubleshooting/ruby.md:78]:
```suggestion
In rare situations, the profiler can trigger [Ruby VM Bug #19991][12] that manifests itself as a "Segmentation fault" with a crash stack trace including the `gc_finalize_deferred` function.
```

**Review by aliciascott — CHANGES_REQUESTED** [review]
Just a quick wording suggestion as we want to avoid words like "we/ we've" , let me know once updated, request re review and I'll get it merged :)

**Comment by ivoanjo** [comment]
Suggestion sounds good, thank you! We're actually going to try to have a workaround in the library for this too, but I think for now let's go with the doc and I may come back later with a tweak that says "on version xx and above the workaround is no longer needed". :)

**Review by aliciascott — APPROVED** [review]
looks good !

**Pull request merged**